Tests/js/views/dashboard/assets/ez-dashboardblockallcontentview-tests.js

```diff
@@ -0,0 +1,265 @@
+
+    renderTest = new Y.Test.Case({
+        name: 'eZ Dashboard All Content Block View render test',
+
+        setUp: function () {
+            this.rootLocation = new Y.Model({
+                locationId: ROOT_LOCATION_ID,
+                pathString: PATH_STRING
+            });
+            this.view = new Y.eZ.DashboardBlockAllContentView(Y.merge(VIEW_CONFIG, {rootLocation: this.rootLocation}));
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+        },
+
+        'Should render view in the loading state': function () {
+            var view = this.view,
+                templateCalled = false,
+                origTpl = view.template,
+                container = view.get('container');
+
+            view.template = function (params) {
+                templateCalled = true;
+
+                Y.Assert.isArray(params.items, 'The `items` variable should be an array');
+                Y.Assert.areSame(0, params.items.length, 'Should not provide any items data');
+                Y.Assert.isFalse(params.loadingError, 'The `loadingError` should not be enabled');
+
+                return origTpl.apply(this, arguments);
+            };
+
+            view.render();
+
+            Y.Assert.isTrue(templateCalled, 'The template should have been used to render view');
+            Y.Assert.isTrue(container.hasClass(CLASS_LOADING), 'Should add the loading state CSS class to the view container');
+        },
+
+        'Should render view with rows': function () {
+            var view = this.view,
+                contentMock = new Y.Mock(),
+                contentTypeMock = new Y.Mock(),
+                locationMock = new Y.Mock(),
+                contentInfoMock = new Y.Mock(),
+                origTpl = view.template,
+                templateCalled = false,
+                list;
+
+            Y.Mock.expect(contentMock, {
+                method: 'toJSON',
+                returns: {}
+            });
+
+            Y.Mock.expect(contentTypeMock, {
+                method: 'toJSON',
+                returns: {}
+            });
+
+            Y.Mock.expect(contentInfoMock, {
+                method: 'toJSON',
+                returns: {}
+            });
+
+            Y.Mock.expect(locationMock, {
+                method: 'toJSON',
+                returns: {}
+            });
+
+            Y.Mock.expect(locationMock, {
+                method: 'get',
+                args: ['contentInfo'],
+                returns: contentInfoMock
+            });
+
+            list = [{
+                content: contentMock,
+                contentType: contentTypeMock,
+                location: locationMock
+            }];
+
+            view.template = function (params) {
+                templateCalled = true;
+
+                Y.Assert.isArray(params.items, 'The `items` variable should be an array');
+                Y.Assert.areSame(list.length, params.items.length, 'Should provide data of 1 item');
```

Comment:
that's not enough, you also have to check that the element in `params.items` is an object and it contains the `content`, `contentInfo`, `location`, `contentType` property with the correct value ie the return values of the `toJSON` methods.